.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,68 @@
+name: Emscripten build and test
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+
+      - name: Install pyodide-build
+        run: |
+          pip install pyodide-build
+          echo EMSCRIPTEN_VERSION=$(pyodide config get emscripten_version) >> $GITHUB_ENV
+
+      - name: Setup Emscripten
+        uses: mymindstorm/setup-emsdk@v12
+        with:
+          version: ${{ env.EMSCRIPTEN_VERSION }}
+
+      - name: Build
+        run: pyodide build
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      # clone awkward and build a wheel, copy the wheel
+      - name: Awkward wheel
+        run: |
+          git clone https://github.com/scikit-hep/awkward.git
+          cd awkward
+          git checkout main
+          git submodule update --init --recursive
```

Comment:
```suggestion
          git submodule update --init --recursive
          pipx run nox -s prepare -- --headers --signatures --tests
```

Reply:
n.b. you'll still need awkward's C++ components too

Reply:
My initial goal was to test the build (only uproot) and run the tests. However the uproot build will only work if the dependencies (awkward) are available, and there is a significant delay between the publication in pypi and it's availability from pyodide, this means that this CI will fail for a while whenever there is a release that modifies dependencies.

To test the CI job I though of building the awkward (and awkward-cpp) wheel in this CI job (in particular the version of awkward listed in pyproject.toml).

I am having some troubles understanding the necessary steps though.